modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/portlet_url.js

```diff
@@ -242,41 +242,16 @@ AUI.add(
 
 				var options = instance.options;
 
-				var reservedParams = instance.reservedParams;
-
-				var resultURL = new A.Url(options.basePortletURL);
-
-				var portletId = reservedParams.p_p_id;
-
-				if (!portletId) {
-					portletId = resultURL.getParameter('p_p_id');
-				}
-
-				var namespacePrefix = Util.getPortletNamespace(portletId);
-
-				A.each(reservedParams, function(item, index) {
-					if (Lang.isValue(item)) {
-						resultURL.setParameter(index, item);
-					}
-				});
-
-				A.each(instance.params, function(item, index) {
-					if (Lang.isValue(item)) {
-						resultURL.setParameter(namespacePrefix + index, item);
-					}
-				});
-
-				if (options.secure) {
-					resultURL.setProtocol('https');
-				}
-
-				var value = resultURL.toString();
-
-				if (options.escapeXML) {
```

Comment:
Are we sure we can drop `options.escapeXML`? This may be something that we want to do always, or not at all. It is pretty insane to think somebody would send HTML as a part of URL.

Reply:
Yes because the `setEscapeXML` method is used only once and it sets the value to false, which in turn does nothing.